### PR TITLE
feat(RELEASE-982): add the catalog (RHEC) url to a results file

### PR DESCRIPTION
**Pull request by happybhati**
This commit adds the catalog (RHEC) url to a results file A task generates a results/publish-pyxis-repository-results.json file with all the catalog url generated.

### CODE REVIEW

File: tasks/publish-pyxis-repository/README.md
Comment (johnbieren) [inline tasks/publish-pyxis-repository/README.md]: need one more space after resultsDirPath

File: tasks/publish-pyxis-repository/README.md
Comment (johnbieren) [inline tasks/publish-pyxis-repository/README.md]: New parameter, let's go to 2.0.0

File: tasks/publish-pyxis-repository/publish-pyxis-repository.yaml
Comment (johnbieren) [inline tasks/publish-pyxis-repository/publish-pyxis-repository.yaml]: This and the following line can be combined into one

File: tasks/publish-pyxis-repository/publish-pyxis-repository.yaml
Comment (johnbieren) [inline tasks/publish-pyxis-repository/publish-pyxis-repository.yaml]: You shouldn't need this. It is done here https://github.com/konflux-ci/release-service-catalog/blob/development/tasks/collect-data/collect-data.yaml#L87

File: tasks/publish-pyxis-repository/publish-pyxis-repository.yaml
Comment (johnbieren) [inline tasks/publish-pyxis-repository/publish-pyxis-repository.yaml]: I'm not sure how this will behave since `images` is used here https://github.com/konflux-ci/release-service-catalog/blob/development/tasks/push-snapshot/push-snapshot.yaml#L96 too. I'd recommend checking the release cr in the e2e to make sure both tasks results are there. The update-cr-status task may overwrite one task's results with the other

File: tasks/publish-pyxis-repository/publish-pyxis-repository.yaml
Comment (johnbieren) [inline tasks/publish-pyxis-repository/publish-pyxis-repository.yaml]: The point of the ticket is for a url like catalog.redhat.com, not starting with the PYXIS_URL

File: tasks/publish-pyxis-repository/tests/test-publish-pyxis-repository.yaml
Comment (johnbieren) [inline tasks/publish-pyxis-repository/tests/test-publish-pyxis-repository.yaml]: Why is this needed now?

**Review by johnbieren — CHANGES_REQUESTED** [review]

File: tasks/publish-pyxis-repository/README.md
Comment (happybhati) [reply tasks/publish-pyxis-repository/README.md]: Fixed!

File: tasks/publish-pyxis-repository/README.md
Comment (happybhati) [reply tasks/publish-pyxis-repository/README.md]: Fixed, thanks!

File: tasks/publish-pyxis-repository/publish-pyxis-repository.yaml
Comment (happybhati) [reply tasks/publish-pyxis-repository/publish-pyxis-repository.yaml]: Done!

File: tasks/publish-pyxis-repository/publish-pyxis-repository.yaml
Comment (happybhati) [reply tasks/publish-pyxis-repository/publish-pyxis-repository.yaml]: Fixed, got it.

File: pipelines/rh-push-to-registry-redhat-io/README.md
Comment (mmalina) [inline pipelines/rh-push-to-registry-redhat-io/README.md]: Which task? This is the pipeline README.

File: pipelines/rh-push-to-registry-redhat-io/README.md
Comment (mmalina) [inline pipelines/rh-push-to-registry-redhat-io/README.md]: This it the popeline README. No parameter was added to the pipeline.

File: pipelines/rh-push-to-registry-redhat-io/rh-push-to-registry-redhat-io.yaml
Comment (mmalina) [inline pipelines/rh-push-to-registry-redhat-io/rh-push-to-registry-redhat-io.yaml:350]: The same should be added in the rh-advisories pipeline, no?

File: tasks/publish-pyxis-repository/publish-pyxis-repository.yaml
Comment (mmalina) [inline tasks/publish-pyxis-repository/publish-pyxis-repository.yaml]: `images` doesn't sound very unique. If we want to make sure it's unique, we should use something like `catalog_urls`.

File: tasks/publish-pyxis-repository/publish-pyxis-repository.yaml
Comment (mmalina) [reply tasks/publish-pyxis-repository/publish-pyxis-repository.yaml]: Agreed with Johnny. The url should be something like "https://catalog.redhat.com/software/containers/rhel9/rhel-bootc/6605573d4dbfe41c3d839c69"

File: tasks/publish-pyxis-repository/publish-pyxis-repository.yaml
Comment (mmalina) [reply tasks/publish-pyxis-repository/publish-pyxis-repository.yaml]: Maybe you actually meant that the `url` inside is unique. Which is not clear from this comment at all. But I did a little experiment and it doesn't work:
```sh
$ JSON1='{"images": [{"name": "comp1", "arches": ["amd64", "arm64"]}, {"name": "comp2", "arches": ["arm64"]}]}'
$ JSON2='{"images": [{"name": "comp1", "url": "catalog.rh"}, {"name": "comp2", "url": "catalog2"}]}'
$ jq "${JSON1} * . " <<< "$JSON2"
{
  "images": [
    {
      "name": "comp1",
      "url": "catalog.rh"
    },
    {
      "name": "comp2",
      "url": "catalog2"
    }
  ]
}
```

File: pipelines/rh-push-to-registry-redhat-io/rh-push-to-registry-redhat-io.yaml
Comment (johnbieren) [reply pipelines/rh-push-to-registry-redhat-io/rh-push-to-registry-redhat-io.yaml:350]: yes

File: tasks/publish-pyxis-repository/publish-pyxis-repository.yaml
Comment (johnbieren) [reply tasks/publish-pyxis-repository/publish-pyxis-repository.yaml]: Resolving in favor of https://github.com/konflux-ci/release-service-catalog/pull/503#discussion_r1711198212

File: tasks/publish-pyxis-repository/publish-pyxis-repository.yaml
Comment (johnbieren) [reply tasks/publish-pyxis-repository/publish-pyxis-repository.yaml]: I resolved my thread about a similar concern. We can keep it all here in this one

File: pipelines/rh-push-to-registry-redhat-io/rh-push-to-registry-redhat-io.yaml
Comment (happybhati) [reply pipelines/rh-push-to-registry-redhat-io/rh-push-to-registry-redhat-io.yaml:350]: Done

File: pipelines/rh-push-to-registry-redhat-io/README.md
Comment (happybhati) [reply pipelines/rh-push-to-registry-redhat-io/README.md]: Fixed

File: pipelines/rh-push-to-registry-redhat-io/README.md
Comment (happybhati) [reply pipelines/rh-push-to-registry-redhat-io/README.md]: ah my bad, fixed!

File: tasks/publish-pyxis-repository/publish-pyxis-repository.yaml
Comment (happybhati) [reply tasks/publish-pyxis-repository/publish-pyxis-repository.yaml]: Yes, I tested using `image`, but it will not work as expected.
Adding a unique name as Martin suggested, `catalog_urls`, but pending test. 
I am not getting a chance to test properly while on PTO. I will continue once I return.

File: tasks/publish-pyxis-repository/publish-pyxis-repository.yaml
Comment (happybhati) [reply tasks/publish-pyxis-repository/publish-pyxis-repository.yaml]: Got it, Thanks.
I am trying to generate a proper catalog URL, and I might need help when I return from PTO to verify and test properly.

File: tasks/publish-pyxis-repository/publish-pyxis-repository.yaml
Comment (mmalina) [inline tasks/publish-pyxis-repository/publish-pyxis-repository.yaml:149]: The base url will be catalog.stage.redhat.com for images pushed to stage. Not sure what the best way to determine that is. Maybe you'll have to get the rh_registry value and add an `if`?

File: tasks/publish-pyxis-repository/publish-pyxis-repository.yaml
Comment (mmalina) [reply tasks/publish-pyxis-repository/publish-pyxis-repository.yaml]: It still doesn't seem correct. Component name is irrelevant here. You need to repo path under registry.redhat.io/.

Also, the base url will be catalog.stage.redhat.com for stage pushes, so we need a way to support that as well - perhaps you'll need to get the rh_registry value and decide based on that.

File: pipelines/rh-push-to-registry-redhat-io/README.md
Comment (mmalina) [reply pipelines/rh-push-to-registry-redhat-io/README.md]: It's better. But why are there two items? Only one thing was added. Can you try to reword/reformat it? You can have a main item and another one under it. But it shouldn't be two at the same level.

(This applies to both pipelines.)

File: tasks/publish-pyxis-repository/publish-pyxis-repository.yaml
Comment (mmalina) [reply tasks/publish-pyxis-repository/publish-pyxis-repository.yaml]: Yeah, I think it should work with `catalog_urls`. But I guess ideally we should make improvements to be able to show the catalog url for each image alongside the other values. (But I'm not sure how this will be shown in the catalog anyway and this is something we can look at later.)

File: pipelines/rh-push-to-registry-redhat-io/README.md
Comment (johnbieren) [reply pipelines/rh-push-to-registry-redhat-io/README.md]: Example
```
## Changes in 0.10.0
- The `push-snapshot` task now gets the `resultsDirPath` parameter from the `collect-data` results
```

File: tasks/publish-pyxis-repository/publish-pyxis-repository.yaml
Comment (johnbieren) [reply tasks/publish-pyxis-repository/publish-pyxis-repository.yaml]: We can make a separate ticket about fixing the jq combination thing in the update-cr-status task (I think it is there) to better union the keys. @happybhati can you create that when you get a chance and link it here?

File: tasks/publish-pyxis-repository/tests/test-publish-pyxis-repository.yaml
Comment (johnbieren) [inline tasks/publish-pyxis-repository/tests/test-publish-pyxis-repository.yaml]: Why? I don't see anything in your change that would lead to needing a data file

File: pipelines/rh-push-to-registry-redhat-io/README.md
Comment (mmalina) [reply pipelines/rh-push-to-registry-redhat-io/README.md]: What Johnny says, except we standardized on using `*` and not `-` for markdown lists :)

File: tasks/publish-pyxis-repository/publish-pyxis-repository.yaml
Comment (happybhati) [reply tasks/publish-pyxis-repository/publish-pyxis-repository.yaml]: Sure, I will create and post the ticket here for the changes in the update-cr-status task. Thanks.

File: pipelines/rh-push-to-registry-redhat-io/README.md
Comment (happybhati) [reply pipelines/rh-push-to-registry-redhat-io/README.md]: Done, thanks!

File: tasks/publish-pyxis-repository/publish-pyxis-repository.yaml
Comment (happybhati) [reply tasks/publish-pyxis-repository/publish-pyxis-repository.yaml]: Okay, I am trying to generate the correct URL. I might reach out to you offline for questions.
Thanks for mentioning that the base URL will change according to stage/ prod.

File: tasks/publish-pyxis-repository/tests/test-publish-pyxis-repository.yaml
Comment (happybhati) [reply tasks/publish-pyxis-repository/tests/test-publish-pyxis-repository.yaml]: My bad. It was from my other local test repo, and I don't know how this got into this change.
Thanks for noticing. I have removed it now.

File: pipelines/rh-advisories/README.md
Comment (johnbieren) [inline pipelines/rh-advisories/README.md:1]: The version numbers here seem off (and the rh-advisories yaml didn't even have the version string bumped)

Comment (happybhati) [reply pipelines/rh-advisories/README.md:1]: Fixed, now merge conflict and rebase messed things up a little bit. :/

File: tasks/publish-pyxis-repository/publish-pyxis-repository.yaml
Comment (happybhati) [reply tasks/publish-pyxis-repository/publish-pyxis-repository.yaml]: I have created this story https://issues.redhat.com/browse/RELEASE-1137 for fixing `jq` combination in `update-cr-status` task

**Comment by happybhati** [comment]
Fixing the README format, I got some redundant code while resolving the conflict and rebasing simultaneously. I will post a comment once it is ready for review.

**Comment by happybhati** [comment]
Ready for review!

**Comment by happybhati** [comment]
test release-pipelines-e2e-suite

File: tasks/publish-pyxis-repository/publish-pyxis-repository.yaml
Comment (mmalina) [reply tasks/publish-pyxis-repository/publish-pyxis-repository.yaml]: It looks ok now.

File: tasks/publish-pyxis-repository/README.md
Comment (mmalina) [inline tasks/publish-pyxis-repository/README.md]: Can you mention it contains links to Red Hat Catalog?

More importantly, this change adds a new parameter (which is what makes it a breaking change), can you add an item for that?

File: tasks/publish-pyxis-repository/publish-pyxis-repository.yaml
Comment (mmalina) [inline tasks/publish-pyxis-repository/publish-pyxis-repository.yaml]: Why the `-n` here? It's completely ok to have a new line at the end here. I would probably go even further and format the json again, so that it's more readable in the logs:
```suggestion
        jq <<< "${RESULTS_JSON}" | tee "$RESULTS_FILE"
```

**Review by mmalina — COMMENTED** [review]
Just two minor comments. Otherwise it looks good.

File: tasks/publish-pyxis-repository/publish-pyxis-repository.yaml
Comment (happybhati) [reply tasks/publish-pyxis-repository/publish-pyxis-repository.yaml]: Make sense, applied suggested changes. Thanks!

File: tasks/publish-pyxis-repository/README.md
Comment (happybhati) [reply tasks/publish-pyxis-repository/README.md]: Yes, I have made the changes you suggested. Thanks

**Review by mmalina — DISMISSED** [review]

**Review by johnbieren — DISMISSED** [review]

**Comment by happybhati** [comment]
I fixed the checkton warning in the latest commit and rebased from development, waiting for e2e to pass before I merge.

@johnbieren and @mmalina, could you approve Agina, please? Thanks!

**Review by johnbieren — APPROVED** [review]

**Review by mmalina — APPROVED** [review]

**Pull request merged**
